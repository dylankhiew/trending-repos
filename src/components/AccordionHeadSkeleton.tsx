import { StyleSheet, View } from 'react-native';

import { COLOR } from '../constants/colorConstants';
import { SPACING } from '../constants/spacingConstants';

export default function AccordionHeadSkeleton(): JSX.Element {
  return (
    <View style={styles.accordionContainer}>
      <View style={styles.accordionAvatar} />
      <View style={styles.accordionTitleArea}>
        <View style={styles.accordionHeader} />
        <View style={styles.accordionTitle} />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  accordionContainer: {
    display: 'flex',
    flexDirection: 'row',
    flex: 1,
    borderBottomColor: COLOR.GRAY100,
    borderBottomWidth: 1,
    marginTop: 21,
  },
  accordionAvatar: {
    height: 32,
    width: 32,
    borderRadius: 50,
    marginHorizontal: SPACING.S_4,
    backgroundColor: COLOR.GRAY100,
  },
  accordionTitleArea: {
    flexDirection: 'column',
    flex: 1,
  },
  accordionHeader: {
    marginBottom: SPACING.S_3,
    height: 10,
    width: 100,
    backgroundColor: COLOR.GRAY100,
    borderRadius: 5,
  },
  accordionTitle: {
    width: 260,
    height: 10,
    backgroundColor: COLOR.GRAY100,
    borderRadius: 5,
  },
});
