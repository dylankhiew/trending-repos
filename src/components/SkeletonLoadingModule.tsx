import * as React from "react";
import { ReactNode } from "react";
import { SafeAreaView, StyleSheet } from "react-native";

interface ErrorModuleProps {
  component: ReactNode;
  amount: number;
}

export default function SkeletonLoadingModule({
  component,
  amount,
}: ErrorModuleProps): JSX.Element {
  const renderComponent = () => {
    return [...Array(amount)].map((_x, i) => (
      <React.Fragment key={`skeleton-loader-${i}`}>{component}</React.Fragment>
    ));
  };

  return (
    <SafeAreaView style={styles.container}>{renderComponent()}</SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
    paddingHorizontal: 48,
  },
});
