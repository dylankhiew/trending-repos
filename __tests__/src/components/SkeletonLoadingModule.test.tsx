import React from 'react';
import renderer from 'react-test-renderer';

import AccordionHeadSkeleton from '../../../src/components/AccordionHeadSkeleton';
import SkeletonLoadingModule from '../../../src/components/SkeletonLoadingModule';

it('should render correctly', () => {
  const component = <AccordionHeadSkeleton />;
  const amount = 10;
  const tree = renderer
    .create(<SkeletonLoadingModule component={component} amount={amount} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
