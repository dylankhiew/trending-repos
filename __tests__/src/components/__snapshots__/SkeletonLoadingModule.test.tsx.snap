// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 48,
    }
  }
>
  <View
    style={
      {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "marginTop": 21,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8E8E8",
          "borderRadius": 50,
          "height": 32,
          "marginHorizontal": 16,
          "width": 32,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "marginBottom": 12,
            "width": 100,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "width": 260,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "marginTop": 21,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8E8E8",
          "borderRadius": 50,
          "height": 32,
          "marginHorizontal": 16,
          "width": 32,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "marginBottom": 12,
            "width": 100,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "width": 260,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "marginTop": 21,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8E8E8",
          "borderRadius": 50,
          "height": 32,
          "marginHorizontal": 16,
          "width": 32,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "marginBottom": 12,
            "width": 100,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "width": 260,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "marginTop": 21,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8E8E8",
          "borderRadius": 50,
          "height": 32,
          "marginHorizontal": 16,
          "width": 32,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "marginBottom": 12,
            "width": 100,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "width": 260,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "marginTop": 21,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8E8E8",
          "borderRadius": 50,
          "height": 32,
          "marginHorizontal": 16,
          "width": 32,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "marginBottom": 12,
            "width": 100,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "width": 260,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "marginTop": 21,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8E8E8",
          "borderRadius": 50,
          "height": 32,
          "marginHorizontal": 16,
          "width": 32,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "marginBottom": 12,
            "width": 100,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "width": 260,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "marginTop": 21,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8E8E8",
          "borderRadius": 50,
          "height": 32,
          "marginHorizontal": 16,
          "width": 32,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "marginBottom": 12,
            "width": 100,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "width": 260,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "marginTop": 21,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8E8E8",
          "borderRadius": 50,
          "height": 32,
          "marginHorizontal": 16,
          "width": 32,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "marginBottom": 12,
            "width": 100,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "width": 260,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "marginTop": 21,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8E8E8",
          "borderRadius": 50,
          "height": 32,
          "marginHorizontal": 16,
          "width": 32,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "marginBottom": 12,
            "width": 100,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "width": 260,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "marginTop": 21,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8E8E8",
          "borderRadius": 50,
          "height": 32,
          "marginHorizontal": 16,
          "width": 32,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "marginBottom": 12,
            "width": 100,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "#E8E8E8",
            "borderRadius": 5,
            "height": 10,
            "width": 260,
          }
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
